lib/archive/archive.go

```diff
@@ -0,0 +1,97 @@
+package archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+func DecompressTarLz4(ctx context.Context, inputFile string, outputDir string) error {
+	// Open the .tar.lz4 file.
+	file, err := os.Open(inputFile)
+	if err != nil {
+		return errors.Wrap(err, "failed to open file")
+	}
+	defer file.Close()
+
+	// Create an LZ4 reader.
+	lz4Reader := lz4.NewReader(file)
+
+	// Decompress into memory or stream directly.
+	var buf bytes.Buffer
+	_, err = io.Copy(&buf, lz4Reader)
+	if err != nil {
+		return errors.Wrap(err, "failed to decompress lz4")
```

Comment:
see go style guide, no need to add "failed" when wrapping

Reply:
Done.